### PR TITLE
Functional Windows implementation

### PR DESCRIPTION
**Pull request by Jon-Salmon**
Adds support for this plugin on Windows via the win32 package.

I'm not convinced what the best way to configure this in wakelock/lib/wakelock.dart as this implementation doesn't use method channels but instead uses the win32 package for the platform implementation.

See this issue for more details: https://github.com/flutter/flutter/issues/52267

### CODE REVIEW

File: wakelock_windows/lib/wakelock_windows.dart [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    if (enable) {
+      SetThreadExecutionState(_ES_CONTINUOUS | _ES_DISPLAY_REQUIRED);
+    } else {
+      SetThreadExecutionState(_ES_CONTINUOUS);
+    }
+    _isEnabled = enable;
+  }
+
+  @override
+  Future<bool> get enabled async => _isEnabled;
```
Comment (creativecreatorormaybenot) [inline wakelock_windows/lib/wakelock_windows.dart:28]: Is there a way to not handle this using a class member on the Dart side and instead directly retrieve the thread execution state from win32?

I would prefer that a lot.

File: wakelock_windows/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+This project is a starting point for a Dart
+[package](https://flutter.dev/developing-packages/),
+a library module containing code that can be shared easily across
+multiple Flutter or Dart projects.
+
+For help getting started with Flutter, view our 
+[online documentation](https://flutter.dev/docs), which offers tutorials, 
+samples, guidance on mobile development, and a full API reference.
+
```
Comment (creativecreatorormaybenot) [inline wakelock_windows/README.md:15]: This is redundant.

```suggestion
```

**Review by creativecreatorormaybenot — CHANGES_REQUESTED** [review]
Thank you so much for opening this PR and contributing @Jon-Salmon ❤️ 

Great work - I like it a lot ✨ 

---

From my side, the way I would proceed is resolving the two comments I have (you should resolve those or respond to them). After that, I think this can be merged. Then I would probably make some minor changes to the descriptions, READMEs, CHANGELOGs, etc. throughout the repo and publish everything 🚀 Does that sound good to you? :)

**Comment by creativecreatorormaybenot** [comment]
About https://github.com/flutter/flutter/issues/52267 - I think this would allow improving both macOS and Windows support a lot (since it has been merged now).

However, I think that it is fine to publish it like this until this reaches `stable` 🙂 Thanks a lot for linking the issue!

File: wakelock_windows/lib/wakelock_windows.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import 'dart:async';
+
+import 'package:wakelock_platform_interface/wakelock_platform_interface.dart';
+import 'package:win32/win32.dart';
+
+const _ES_CONTINUOUS = 0x80000000;
```
Comment (creativecreatorormaybenot) [inline wakelock_windows/lib/wakelock_windows.dart:6]: I think we can write docs linking to https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-setthreadexecutionstate#parameters for reference :)

I would also add `pedantic` and enforce docs, however, you do not need to worry about it - I can do that once this has been merged 👍

Comment (creativecreatorormaybenot) [reply wakelock_windows/lib/wakelock_windows.dart:6]: (so this is not blocking the PR from being merged)

File: wakelock_windows/lib/wakelock_windows.dart (hunk 1 above)
Comment (Jon-Salmon) [reply wakelock_windows/lib/wakelock_windows.dart:36]: Unfortunately, this not supported by the Windows API. I have however added some validation so the internal state will not change if the win32 API call fails.

File: wakelock_windows/README.md (hunk 2 above)
Comment (Jon-Salmon) [reply wakelock_windows/README.md]: Good spot

File: wakelock_windows/lib/wakelock_windows.dart (hunk 3 above)
Comment (Jon-Salmon) [reply wakelock_windows/lib/wakelock_windows.dart:6]: I've added some additional docs + pedantic with some extra configs. Feel free to change this if it's not the correct configuration for your liking.

**Comment by Jon-Salmon** [comment]
Thanks for looking over this, I've made all the fixes I think are appropriate. Feel free to change anything you wish to before publishing.

**Comment by creativecreatorormaybenot** [comment]
Thanks so much for this! I will make sure to get this published today.

**Comment by creativecreatorormaybenot** [comment]
@Jon-Salmon Hi again :) I am wondering: do you have Twitter?

I would like to post about the added Windows support and attribute the feature to you.

**Comment by Jon-Salmon** [comment]
@creativecreatorormaybenot I'm afraid I don't but feel free to post anyway.

**Pull request merged**